NHC as a Charm Resource

omnivector-kelly opened the pull request:
**Please provide description of the purpose of this pull request, as well as its
motivation and context.**

Is it a new feature? A bug fix? Does it address an existing issue?

- This provides NHC via a resource instead of downloading the resource in ops manager. 

**How was the code tested?**
make tests

Final checklist
- [x] I am the author of these changes, or I have the rights to submit them.
- [x] I added the relevant changed to the README and/or documentation.
- [x] I self reviewed my own code.
- [x] I updated the `CHANGELOG` according to the [contributing
  guide](https://omnivector-solutions.github.io/osd-documentation/master/contributing.html#changelog)

omnivector-kelly committed:
Remove unnecessary resources from charms metadata.yaml

omnivector-kelly committed:
Remove unnecessary resources from charms metadata.yaml

omnivector-kelly committed:
Remove unnecessary resources from charms metadata.yaml

omnivector-kelly committed:
Remove unnecessary resources from charms metadata.yaml

omnivector-kelly committed:
Fix F strings in slurmd/charm.py

omnivector-kelly committed:
Fix F strings in slurmd/charm.py

heitorPB commented on charm-slurmd/src/charm.py:
```diff
@@ -105,6 +105,25 @@ def __init__(self, *args):
+        f = open(nhc_tmp_file, 'wb')
+        f.write(content)
+        f.close()
+        logger.debug(f"### Relocating NHC Resource Successful")
```
```suggestion
```

This is not needed, the variable `nhc_path` will contain a valid path for the resource.

I suggest you pass it to the `successful_installation = self._slurm_manager.install(custom_repo, nhc_path)` line below and extract it in the slurm-ops-manager code.

omnivector-kelly committed:
Update Changelog

omnivector-kelly committed:
Update Changelog

omnivector-kelly committed:
Add NHC and resources to slurm-charms metadata.yaml.

omnivector-kelly committed:
Add NHC and resources to slurm-charms metadata.yaml.

omnivector-kelly committed:
Pass nhc_path down to slurm_manager install metadata.yaml.

omnivector-kelly committed:
Pass nhc_path down to slurm_manager install metadata.yaml.

omnivector-kelly committed:
fix blank lines

omnivector-kelly committed:
fix blank lines

omnivector-kelly committed:
Remove NHC version from slurmd charm.py

omnivector-kelly committed:
Remove NHC version from slurmd charm.py

omnivector-kelly committed:
Change slurm-ops-manager version

omnivector-kelly committed:
Change slurm-ops-manager version

jaimesouza committed:
slurmd: add an action to install singularity

Add an action on charm-slurmd to install singularity using the
resources singularity-deb and singularity-rpm, both also included
in this commit.

The .deb (singularity-ce_3.10.2-focal_amd64.deb) and
.rpm (singularity-ce-3.10.2-1.el7.x86_64.rpm) file should be
downloaded from Singulary Github Releases.

Note: The charm-slurmd/requirements.txt must be updated to use
the version of slurm-ops-manager with the Singulary class
implemented, before applying the action.

Co-authored-by: Heitor <heitorpbittencourt@gmail.com>

omnivector-kelly committed:
Merge branch 'dev-kelly-third' of github.com:omnivector-solutions/slurm-charms into dev-kelly-third
